### ISSUE TITLE
1.0.14 tarball contains older generated files

### ISSUE THREAD

**Issue by tagoh** [fribidi/fribidi#207]
Someone reported to Fedora that FriBiDi changed the behavior after upgrading to 1.0.14: https://bugzilla.redhat.com/show_bug.cgi?id=2279842

The detailed reproducible is available here (inserting a space between `$` and `(` for some line in snippet to avoid unexpected formatting btw):
https://bugzilla.redhat.com/show_bug.cgi?id=2279842#c1

> This is a bug in fribidi-1.0.14. A reproducer:
>
> `$ cat test.c `
> ```
> #include <fribidi-types.h>
> #include <fribidi-mirroring.h>
> #include <stdio.h>
> 
> int main(void) {
>     FriBidiChar input = 0x2BFE;
>     FriBidiChar output;
>     fribidi_boolean ret;
>     
>     ret = fribidi_get_mirror_char(input, &output);
>     printf("input=0x%jx, output=0x%jx, fribidi_get_mirror_char() retval=%u\n", input, output, ret);
> 
>     return 0;
> }
> ```
> 
> $ gcc $ (pkg-config --cflags fribidi) test.c $(pkg-config --libs fribidi)
> 
> fribidi-1.0.13 correctly finds that U+2BFE mirrors to U+221F:
> 
> $ rpm -q fribidi
> fribidi-1.0.13-4.fc40.x86_64
> $ ./a.out 
> input=0x2bfe, output=0x221f, fribidi_get_mirror_char() retval=1
> 
> However, after upgrading to 1.0.14, fribidi_get_mirror_char() thinks that U+2BFE has not mirroring chacter:
> 
> $ rpm -q fribidi
> fribidi-1.0.14-1.fc41.x86_64
> $ ./a.out 
> input=0x2bfe, output=0x2bfe, fribidi_get_mirror_char() retval=0
> 
> I compared Unicode data sets from <https://www.unicode.org/Public/14.0.0/ucd/BidiMirroring.txt> and <https://www.unicode.org/Public/15.1.0/ucd/BidiMirroring.txt> and there is no change:
> 
> $ grep 2BFE BidiMirroring.txt-1*
> BidiMirroring.txt-14.0.0:221F; 2BFE # RIGHT ANGLE
> BidiMirroring.txt-14.0.0:2BFE; 221F # REVERSED RIGHT ANGLE
> BidiMirroring.txt-15.1.0:221F; 2BFE # RIGHT ANGLE
> BidiMirroring.txt-15.1.0:2BFE; 221F # REVERSED RIGHT ANGLE

**Comment by tagoh** [comment]
Here looks very weird and didn't really help but... just to make sure:

**Prerequisites:**
1. Building the above test case first as a.out
1. Prepare test script like:
```
#! /bin/sh
meson setup --reconfigure build
meson compile -C build
LD_LIBRARY_PATH=./build/libs ./a.out
exit $?
```

> $ git bisect start
> status: waiting for both good and bad commits
> $ git bisect bad
> status: waiting for good commit(s), bad commit known
> $ git bisect good v1.0.13
> Bisecting: 2 revisions left to test after this (roughly 2 steps)
> [3002ffea02581b514d70c56d972af12082c4e70c] Update test cases to Unicode 15.1
> $ git bisect run ./test.sh
> <snip>
> bca04dc3cd3af85a9d9220c430737333634d622a is the first bad commit
> commit bca04dc3cd3af85a9d9220c430737333634d622a
> Author: Dov Grobgeld <dov.grobgeld@gmail.com>
> Date:   Thu Apr 25 06:44:01 2024 +0300
> 
>     Bumped version to 1.0.14
> 
>  NEWS         | 5 +++++
>  configure.ac | 2 +-
>  meson.build  | 2 +-
>  3 files changed, 7 insertions(+), 2 deletions(-)
> bisect found first bad commit

**Comment by tagoh** [comment]
Okay, finally tracked this down.  tarball from https://github.com/fribidi/fribidi/releases/download/v1.0.14/fribidi-1.0.14.tar.xz contains some .tab.i files which is supposed to be generated at the build time and they are apparently older and based on 10.0.0 somehow.

As well as lib/fribidi-unicode-version.h.

This works as expected after removing them.

**Comment by dov** [comment]
Hi @tagoh . Great catch! I'll prepare a fresh release in the next few days to ensure that there are no remnant i files.

**Comment by dov** [comment]
I have just released v1.0.15 in which I have fixed this.

Curiously I could not get rid of the stall include files by automake until I cloned a clean repo and did `make dist` there. I'd be happy to receive a PR that fixes this.

Or perhaps we should drop automake support alltogether and only support meson. Is there any reason not to do that?

**Comment by tagoh** [comment]
I didn't try though, `make maintainer-clean` may works?

Or you could try publishing a release through a pipeline in GitHub Action. they should be always clean and reproducible.

For dropping autotools support, I have no objection for that. we, Fedora has already migrated to mison to build fribidi. I have no idea for other distros.
